azure/services/managedclusters/client.go

```diff
@@ -156,12 +156,7 @@ func (ac *azureClient) IsDone(ctx context.Context, future azureautorest.FutureAP
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "managedclusters.azureClient.IsDone")
 	defer done()
 
-	isDone, err := future.DoneWithContext(ctx, ac.managedclusters)
-	if err != nil {
-		return false, errors.Wrap(err, "failed checking if the operation was complete")
-	}
-
-	return isDone, nil
+	return future.DoneWithContext(ctx, ac.managedclusters)
```

Comment:
I imagine this is a change we'd make to all the service clients at once. I'll only include the change to managedclusters for now and once we decide how these should look then I'll make any necessary changes to the others.

Reply:
Updated the rest.